### PR TITLE
fix(hpa): unable to target a deployment with a container that has volume mounts

### LINKED ISSUE #1423: HPA - Typescript - Maximum call stack size exceeded
With the following code HPA got stuck in infinite recursion:

``` typescript
const app = new App();
const chart = new Chart(app, 'test', {namespace: 'test'});
const deployment = new Deployment(chart, 'deployment');
const volumeMounts = [{ volume: Volume.fromEmptyDir(deployment, 'test-volume','test'), path: './test'}];

deployment.addContainer({
  image: 'ubuntu',
  name: 'test',
  volumeMounts,
});

new HorizontalPodAutoscaler(chart, 'HPA', {
  target: deployment,
  maxReplicas: 5,
});
app.synth();
```

error message:

```
(node:1258041) UnhandledPromiseRejectionWarning: RangeError: Maximum call stack size exceeded
    at Function.[Symbol.hasInstance] (<anonymous>)
    at resolve (/home/<strip>/node_modules/cdk8s/src/_resolve.ts:10:12)
    at resolve (/home/<strip>/node_modules/cdk8s/src/_resolve.ts:32:17)
    at resolve (/home/<strip>/node_modules/cdk8s/src/_resolve.ts:32:17)
    at resolve (/home/<strip>/node_modules/cdk8s/src/_resolve.ts:32:17)
    at resolve (/home/<strip>/node_modules/cdk8s/src/_resolve.ts:32:17)
    at resolve (/home/<strip>/node_modules/cdk8s/src/_resolve.ts:32:17)
    at resolve (/home/<strip>/node_modules/cdk8s/src/_resolve.ts:32:17)
    at resolve (/home/<strip>/node_modules/cdk8s/src/_resolve.ts:32:17)
    at resolve (/home/<strip>/node_modules/cdk8s/src/_resolve.ts:32:17)
    at emitUnhandledRejectionWarning (internal/process/promises.js:168:15)
    at processPromiseRejections (internal/process/promises.js:247:11)
    at processTicksAndRejections (internal/process/task_queues.js:96:32)
```

Without volumeMounts it works perfectly.

### PR DESCRIPTION
**Pull request by iliapolo**
The `_toKube` method of `HorizontalPodAutoscaler` was populating the result with a reference to the `.containers` property of a deployment. This created a resolving loop since `containers` can contain constructs, which refer contain circular references to each other via the `host` and `defaultChild` properties. 

Remove the `.containers` from the result of `_toKube`, it was oversight, and is not necessary. In general as a rule of thumb, `_toKube` methods should always result in a JSON serializable object.

Fixes https://github.com/cdk8s-team/cdk8s-plus/issues/1423

### CODE REVIEW

File: src/horizontal-pod-autoscaler.ts [hunk 1]
```diff
@@ -336,10 +336,15 @@ export class HorizontalPodAutoscaler extends Resource {
    * @internal
    */
   public _toKube(): k8s.HorizontalPodAutoscalerSpecV2 {
+    const scalingTarget = this.target.toScalingTarget();
     return {
       maxReplicas: this.maxReplicas,
       minReplicas: this.minReplicas,
-      scaleTargetRef: this.target.toScalingTarget(),
+      scaleTargetRef: {
```
Comment (iliapolo) [inline src/horizontal-pod-autoscaler.ts:343]: This thing was mistakenly passing the `.containers` property from `scalingTarget`, which doesn't belong here.

**Review by ryparker — APPROVED** [review]

**Comment by ryparker** [comment]
Do we need backport PRs?

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-24/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/1444)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Comment by iliapolo** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|k8s-23/main|[](https://github.com/cdk8s-team/cdk8s-plus/pull/1445)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**